apps/deploy-web/src/components/new-deployment/ManifestEdit.tsx

```diff
@@ -257,30 +295,46 @@ export const ManifestEdit: React.FunctionComponent<Props> = ({ editedManifest, s
         </div>
       </div>
 
-      <div className="mb-2 flex items-center">
-        <Button
-          variant={selectedSdlEditMode === "builder" ? "default" : "outline"}
-          onClick={() => onModeChange("builder")}
-          size="sm"
-          className="flex-grow rounded-e-none sm:flex-grow-0"
-          disabled={!!parsingError && selectedSdlEditMode === "yaml"}
-        >
-          Builder
-        </Button>
-        <Button
-          variant={selectedSdlEditMode === "yaml" ? "default" : "outline"}
-          color={selectedSdlEditMode === "yaml" ? "secondary" : "primary"}
-          onClick={() => onModeChange("yaml")}
-          size="sm"
-          className="flex-grow rounded-s-none sm:flex-grow-0"
-        >
-          YAML
-        </Button>
+      <div className="mb-2 flex gap-2">
+        <div className="flex items-center">
+          <Button
+            variant={selectedSdlEditMode === "builder" ? "default" : "outline"}
+            onClick={() => onModeChange("builder")}
+            size="sm"
+            className="flex-grow rounded-e-none sm:flex-grow-0"
+            disabled={!!parsingError && selectedSdlEditMode === "yaml"}
+          >
+            Builder
+          </Button>
+          <Button
+            variant={selectedSdlEditMode === "yaml" ? "default" : "outline"}
+            color={selectedSdlEditMode === "yaml" ? "secondary" : "primary"}
+            onClick={() => onModeChange("yaml")}
+            size="sm"
+            className="flex-grow rounded-s-none sm:flex-grow-0"
+          >
+            YAML
+          </Button>
+        </div>
+        {!templateId && (
+          <>
+            <input type="file" ref={fileUploadRef} onChange={propagateUploadedSdl} style={{ display: "none" }} accept=".yml,.yaml,.txt" />
+            <Button
+              variant="outline"
+              color="secondary"
+              onClick={() => triggerFileInput()}
+              size="sm"
+              className="flex-grow hover:bg-primary hover:text-white sm:flex-grow-0"
+            >
+              Upload SDL
+            </Button>
+          </>
+        )}
       </div>
 
       {parsingError && <Alert variant="warning">{parsingError}</Alert>}
 
-      {selectedSdlEditMode === "yaml" && (
+      {selectedSdlEditMode === "yaml" && editedManifest && (
```

Comment:
I think we can render the editor even if there's no `editedManifest`. Users can just copy paste an external SDL without going through the upload button.